### PR TITLE
new module for merging tables output

### PR DESCRIPTION
**Pull request by danilodileo**
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/metatdenovo/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/metatdenovo _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

I modified a bit `collect_stats.nf` module. Now is able to take another table as input. this table contains some important info as the database used for the annotation and, for each samples, the number of orfs that have at least one read mapping to them.

### CODE REVIEW

File: modules/local/collect_stats.nf [hunk 1]
```diff
@@ -41,6 +40,18 @@ process COLLECT_STATS {
         read_trimlogs = "%>%"
     }
 
+    if (mergetab) {
+        read_mergetab = """
+
+        mergetab <- list.files(pattern = "*_merged_table.tsv" ) %>%
+            map_df(~read_tsv(.,  show_col_types  = TRUE))
```
Comment (erikrikarddaniel) [inline modules/local/collect_stats.nf]:
```suggestion
            map_df(~read_tsv(.,  show_col_types  = FALSE))
```

File: modules/local/collect_stats.nf [hunk 2]
```diff
@@ -90,17 +101,20 @@ process COLLECT_STATS {
...
 
+    # Add in stats from taxonomy and function
+    ${read_mergetab}
+
     # Write the table in wide format
     t %>%
         mutate(m = parse_factor(m, levels = TYPE_ORDER, ordered = TRUE)) %>%
         arrange(sample, m) %>%
         pivot_wider(names_from = m, values_from = v) %>%
+        left_join(mergetab) %>%
```
Comment (erikrikarddaniel) [inline modules/local/collect_stats.nf]:
```suggestion
        left_join(mergetab, by = 'NN') %>%
```
Change `NN` to whatever field it is. Avoids having distracting messages in the log.

File: modules/local/merge_summary_tables.nf [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    """
+    #!/usr/bin/env Rscript
+
+    library(data.table)
+    library(dtplyr)
```
Comment (erikrikarddaniel) [inline modules/local/merge_summary_tables.nf]:
```suggestion
```

File: modules/local/sum_eggnog.nf [hunk 4]
```diff
@@ -14,8 +14,8 @@ process SUM_EGGNOG {
 
     output:
 
-    path "eggnog_summary.tsv" , emit: eggnog_summary
-    path "versions.yml"       , emit: versions
+    tuple val(meta), path("eggnog_summary.tsv") , emit: eggnog_summary
```
Comment (erikrikarddaniel) [inline modules/local/sum_eggnog.nf]:
```suggestion
    tuple val(meta), path("${meta.id}.eggnog_summary.tsv") , emit: eggnog_summary
```
? And the corresponding change when you write the file of course.

File: modules/local/sum_eggnog.nf [hunk 5]
```diff
@@ -33,14 +33,11 @@ process SUM_EGGNOG {
...
-    TYPE_ORDER = c('sample', 'database', 'field', 'value')
     # call the tables into variables
-    eggnog <- read_tsv("eggnogs.tsv.gz", show_col_types = FALSE ) %>%
-        as_tibble()
+    eggnog <- read_tsv("eggnogs.tsv.gz", show_col_types = FALSE )
 
     counts <- list.files(pattern = "*_counts.tsv.gz") %>%
-        map_df(~read_tsv(.,  show_col_types  = TRUE)) %>%
-        as_tibble()
+        map_df(~read_tsv(.,  show_col_types  = TRUE))
```
Comment (erikrikarddaniel) [inline modules/local/sum_eggnog.nf]:
```suggestion
        map_df(~read_tsv(.,  show_col_types  = FALSE))
```

File: modules/local/sum_taxonomy.nf [hunk 6]
```diff
@@ -31,15 +31,14 @@ process SUM_TAXONOMY {
     library(readr)
     library(tidyr)
     library(stringr)
+    library(tidyverse)
```
Comment (erikrikarddaniel) [inline modules/local/sum_taxonomy.nf]:
```suggestion
    library(purrr)
```
I'm not sure which packages you actually need, but I think `read_df()` is from `purrr`.

File: modules/local/sum_taxonomy.nf [hunk 7]
```diff
@@ -31,15 +31,14 @@ process SUM_TAXONOMY {
...
     TYPE_ORDER = c('sample', 'database', 'field', 'value')
     # call the tables into variables
-    taxonomy <- read_tsv("${prefix}_taxonomy_classification.tsv", show_col_types = FALSE ) %>%
-        as_tibble()
+    taxonomy <- read_tsv("${prefix}_taxonomy_classification.tsv", show_col_types = FALSE )
 
     counts <- list.files(pattern = "*_counts.tsv.gz") %>%
-        map_df(~read_tsv(.,  show_col_types  = TRUE)) %>%
-        as_tibble()
+        map_df(~read_tsv(.,  show_col_types  = TRUE))
```
Comment (erikrikarddaniel) [inline modules/local/sum_taxonomy.nf]:
```suggestion
        map_df(~read_tsv(.,  show_col_types  = FALSE))
```

**Review by erikrikarddaniel — COMMENTED** [review]
Looks good, but is both `COLLECT_STATS` and `MERGE_SUMMARY_TABLE` needed? Couldn't the latter be incorporated into the former?

**Comment by danilodileo** [comment]
> Looks good, but is both `COLLECT_STATS` and `MERGE_SUMMARY_TABLE` needed? Couldn't the latter be incorporated into the former?

Yes. I started by making the simplest module to get it to work. I can try to have `MERGE_SUMMARY_TABLE` in `COLLECT_STATS`

**Pull request merged**